Dynavity/Dynavity/model/canvas/OnlineCanvas.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+import SwiftUI
+
+class OnlineCanvas: Canvas {
+    static let myUserId = UIDevice.current.identifierForVendor?.uuidString
+            ?? "NO_USER_ID"
+
+    let id: UUID
+    let ownerId: String
+    var shareableId: String {
+        "\(ownerId)/\(id)"
```

Comment:
This should work in theory but by that logic ID should no longer be necessary in storage either (including `CanvasDTO`), since canvas name is consistent across devices and is unique.

Significantly more work will be needed to remove IDs everywhere (again), which should probably be done in another PR.

Reply:
Hmm the reason why there's an ID in CanvasDTO is to support renaming of canvases. I'm not sure if it would work if we remove ID from CanvasDTO

> Significantly more work will be needed to remove canvas IDs (again), which should probably be done in another PR.
Yep, ok!

Reply:
Does this approach not work? https://github.com/Dynavity/dynavity/pull/76/files#diff-860904f80aae08fea7cc64af7e8226cb3d4ce2a70ef18359ef0207c3bbc16059R52

I rewrote the replace function to simply combine a delete and save operation. This allows it to select the appropriate methods based on whether the canvas is an `OnlineCanvas` or not.

Reply:
Oh, right. That should work